contrib/IECoreArnold/src/IECoreArnold/MeshAlgo.cpp

```diff
@@ -265,37 +185,25 @@ AtNode *convertCommon( const IECore::MeshPrimitive *mesh )
 	variablesToConvert.erase( "P" ); // These will be converted
 	variablesToConvert.erase( "N" ); // outside of this function.
 
-	// Convert and remove indexed UVs first. We must perform
-	// the iteration to find the names separately to the iteration
-	// to convert them, because convertIndexedUVSet() removes items
-	// from variablesToConvert, and would therefore invalidate
-	// the interators we were using if we were to do it in one loop.
-	vector<string> uvSetNames;
-	for( PrimitiveVariableMap::iterator it = variablesToConvert.begin(), eIt = variablesToConvert.end(); it != eIt; ++it )
-	{
-		if( boost::ends_with( it->first, "Indices" ) )
-		{
-			uvSetNames.push_back( it->first == "stIndices" ? "" : it->first.substr( 0, it->first.size() - 7 ) );
-		}
-	}
-	for( vector<string>::const_iterator it = uvSetNames.begin(), eIt = uvSetNames.end(); it != eIt; ++it )
-	{
-		convertIndexedUVSet( *it, variablesToConvert, result );
-	}
-
-	// Then convert and remove non-indexed uvs. As above, we must
-	// do this in two phases.
-	uvSetNames.clear();
+	// Find all UV sets. We must perform the iteration to find the
+	// names separately to the iteration to convert them, because
+	// convertUVSet() removes items from variablesToConvert, and
+	// would therefore invalidate the interators we were using if
+	// we were to do it in one loop.
+	vector<PrimitiveVariableMap::iterator> uvSets;
 	for( PrimitiveVariableMap::iterator it = variablesToConvert.begin(), eIt = variablesToConvert.end(); it != eIt; ++it )
 	{
-		if( it->first == "s" || boost::ends_with( it->first, "_s" ) )
+		/// \todo: add a role enum to PrimitiveVariable, so we can distinguish between UVs and
```

Comment:
I think a role _and_ a GeometricInterpretation is probably a bit much. I'm still keen on just moving GeometricInterpretation to PrimitiveVariable, and adding a UV enum value to it. I think this will be much simpler.

Reply:
Oh, that's what I intended anyways. Sorry the comment was unclear.

Reply:
Should I maybe just push a new value on the GeometricInterpretation enum now and address these todos that I've scattered across the renderer backends and dcc converters? Then in a future PR we can address moving it to PrimitiveVariable and whatnot.

Reply:
Up to you - this PR is already huuge though, so I'd be quite happy to just leave this as a todo rather than introduce any further changes...

Reply:
Addressed in 60097586082178017c5b1ec5fa18b9d5a73eb485 and 8e83cdd0701ae499546cf3f1512a86ab4a6e5908